### ISSUE TITLE
Hierarchical Visualization not lining up

### ISSUE THREAD

**Issue by mark-seamlessiot** [MaartenGr/BERTopic#1021]
```
from bertopic import BERTopic
from sklearn.datasets import fetch_20newsgroups
from scipy.cluster import hierarchy as sch

docs = fetch_20newsgroups(subset='all',  remove=('headers', 'footers', 'quotes'))["data"]
topic_model = BERTopic(verbose=True)
topics, probs = topic_model.fit_transform(docs)

linkage_function = lambda x: sch.linkage(x, 'single', optimal_ordering=True)
hierarchical_topics = topic_model.hierarchical_topics(docs, linkage_function=linkage_function)

fig = topic_model.visualize_hierarchy(hierarchical_topics=hierarchical_topics)

fig.show()
```

![image](https://user-images.githubusercontent.com/49913799/220122597-b59cd5ba-f3d7-4da6-97b0-9ee35f242a27.png)

Tested this using my own data and noticed the topic names that show up in the hoverbox dont match the label. I switched to the newsgroup documents to check and it is doing the same thing. It also seems like things arent lining up correctly in terms of merged topic names. In my example, two end topics that both involve carbon have a shared parent topic that ends up being something totally different.

Any ideas whats going on here?

**Comment by MaartenGr** [comment]
Which version of BERTopic are you currently using? Also, can you share the versions of your dependencies? I thought I had this fixed in the newest release but it might not have been the case.

**Comment by mark-seamlessiot** [comment]
<html>
<body>


Cython | 0.29.33 | 0.29.33
-- | -- | --
Pillow | 9.4.0 | 9.4.0
PyYAML | 6.0 | 6.0
bertopic | 0.14.0 | 0.14.0
certifi | 2022.12.7 | 2022.12.7
charset-normalizer | 3.0.1 | 3.0.1
click | 8.1.3 | 8.1.3
colorama | 0.4.6 | 0.4.6
filelock | 3.9.0 | 3.9.0
hdbscan | 0.8.29 | 0.8.29
huggingface-hub | 0.12.1 | 0.12.1
idna | 3.4 | 3.4
importlib-metadata | 6.0.0 | 6.0.0
joblib | 1.2.0 | 1.2.0
llvmlite | 0.39.1 | 0.39.1
nltk | 3.8.1 | 3.8.1
numba | 0.56.4 | 0.56.4
numpy | 1.23.5 | 1.24.2
packaging | 23.0 | 23.0
pandas | 1.5.3 | 1.5.3
pip | 21.3.1 | 23.0.1
plotly | 5.13.0 | 5.13.0
pynndescent | 0.5.8 | 0.5.8
python-dateutil | 2.8.2 | 2.8.2
pytz | 2022.7.1 | 2022.7.1
regex | 2022.10.31 | 2022.10.31
requests | 2.28.2 | 2.28.2
scikit-learn | 1.2.1 | 1.2.1
scipy | 1.10.1 | 1.10.1
sentence-transformers | 2.2.2 | 2.2.2
sentencepiece | 0.1.97 | 0.1.97
setuptools | 60.2.0 | 67.3.2
six | 1.16.0 | 1.16.0
tenacity | 8.2.1 | 8.2.1
threadpoolctl | 3.1.0 | 3.1.0
tokenizers | 0.13.2 | 0.13.2
torch | 1.13.1 | 1.13.1
torchvision | 0.14.1 | 0.14.1
tqdm | 4.64.1 | 4.64.1
transformers | 4.26.1 | 4.26.1
typing-extensions | 4.5.0 | 4.5.0
umap-learn | 0.5.3 | 0.5.3
urllib3 | 1.26.14 | 1.26.14
wheel | 0.37.1 | 0.38.4


</body>
</html>

**Comment by mark-seamlessiot** [comment]
I was using slightly different versions the first time I noticed the issue and ended up upgrading a lot of things to see if that would fix the problem. It is still occuring.

On a side note, after I upgraded a bunch of these, I noticed things ran significantly slower using the same code and data. I have no idea which ones were upgraded or what previous versions were being used but would you have any idea why this might be? before it took less than a minute and now its above 5..

**Comment by MaartenGr** [comment]
> I was using slightly different versions the first time I noticed the issue and ended up upgrading a lot of things to see if that would fix the problem. It is still occurring.

Based on your dependencies, I cannot seem to find what exactly is going wrong here. I thought I had fixed the issue before but it still seems to be happening. I expect it has something to do with #923, perhaps changing the lines as mentioned [here](https://github.com/MaartenGr/BERTopic/issues/923#issuecomment-1386709572) might solve the issue but I had no confirmation yet that it worked for everyone. 

> On a side note, after I upgraded a bunch of these, I noticed things ran significantly slower using the same code and data. I have no idea which ones were upgraded or what previous versions were being used but would you have any idea why this might be? before it took less than a minute and now its above 5..

It might be that you upgraded pytorch to a cpu-version instead of a gpu-version, thereby slowing down inference at the embedding stage.
